### ISSUE TITLE
Click on Template file -> Create Document

### ISSUE THREAD

**Issue by tbsbdr** [ONLYOFFICE/DocumentServer#2839]
### This issue is unique.

- [X] I have used the [search tool](https://github.com/ONLYOFFICE/DocumentServer/issues?q=) and did not find an issue describing my idea.

## Userstory
> As a user I want to create a document by clicking on a template file so that I can use the templates from a shared folder.

## Flow
( pasting here @ibnpetr 's proposal: )
1.	When interacting with a DOTX file (double-clicking, selecting an item in the context menu, hidden opening, or any other logic for opening from a template), a DOCX file without content should be “created.”
2.	The storage (OCIS) should determine the location of the DOCX file: the current or root user folder, or any alternative folder for opening from templates.
3.	It is important that this file can be manipulated like a regular file (CheckFileInfo, Lock, PutFile) except for GetFile. 
Instead of GetFile, a link to the DOTX template should be sent in the CheckFileInfo response (TemplateSource field).
4.	Immediately open this file for editing. OO Docs will convert the template to DOCX and place the DOCX file in the storage using PutFile.
5.	Further editing will proceed as with a regular DOCX file.
 
**Note:**
- Editing a DOTX file as a template is not covered by this scheme.
- Editing from DOTX to DOTX is enabled through a setting in the configuration file.

**Comment by butonic** [comment]
Do I understand correctly that we create an empty DOCX file to a location writeable by the user and fill a new `TemplateSource` property in the GetFileInfo response with a URL to the DOTX template? onlyoffice would detect that new parameter, fetch the template and fill the DOCX file accordingly?

**Comment by ibnpetr** [comment]
> Do I understand correctly that we create an empty DOCX file to a location writeable by the user and fill a new `TemplateSource` property in the GetFileInfo response with a URL to the DOTX template? onlyoffice would detect that new parameter, fetch the template and fill the DOCX file accordingly?

Yes, you can either create a new empty file or copy the template file itself, which is an alternative method used by other vendors.
A new file will be created from the TemplateSource file.

**Comment by micbar** [comment]
After discussion with oC colleagues, we have a vague idea how to implement that.

We should use the app provider /app/new Action to create the new file in the personal space of the user with a server choosen name (File name and user-friendly datetime). Then we can open that via /app/open with the selected template in the wopi CheckFileInfo response.

**Comment by dj4oC** [comment]
@micbar any progress on that item?

**Comment by micbar** [comment]
not yet.

**Comment by micbar** [comment]
@ibnpetr @LuckySpecialist @igwyd  I have an implementation and was hitting a roadblock.

I implemented the workflow as far as Step 3 in the top Post:

- Opening a new empty file and sending a File Download Url like `https://wopiserver/wopi/files/<fileid>/contents` in the `TemplateSource` property.

My Observation:

OnlyOffice tries to download that file, but it doesn't send the Access Token.

Did i miss anyting?

I am using `onlyoffice/4testing-documentserver:8.2.0.108`

**Comment by micbar** [comment]
@alexonlyoffice Thank you. I will try to do an implementation which could work in that context.
